.github/workflows/generate-changelog.yaml

```
@@ -0,0 +1,127 @@
+        description: "The GitHub token for committing the changes."
+        required: true
+
+jobs:
+  read-config-file:
+    name: Read config file
+    runs-on: ubuntu-22.04
+    outputs:
+      content: ${{ steps.read-config.outputs.content }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          # required for getting all tags
+          fetch-depth: 0
+          # required for bypassing status checks when pushing
+          token: "${{ secrets.github-token }}"
+
+      - name: Read changelog config
+        id: read-config
+        # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+        run: |
+          content=$(cat "${{ inputs.config }}")
+          { echo "content<<EOF"; echo "$content"; echo "EOF"; } >> "$GITHUB_OUTPUT"
+        shell: bash
+
+  generate-changelog:
+    name: Generate Changelog
+    runs-on: ubuntu-22.04
+    needs: [read-config-file]
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          # required for getting all tags
+          fetch-depth: 0
+          # required for bypassing status checks when pushing
+          token: "${{ secrets.github-token }}"
+
+      - name: Generate Changelog
+        id: generate-release-changelog
+        uses: heinrichreimer/github-changelog-generator-action@v2.2
+        with:
+          token: "${{ secrets.github-token }}"
+          sinceTag: "${{ inputs.old-tag }}"
+          output: "changes.md"
+          stripGeneratorNotice: "true"
+          issues: "true"
+          unreleased: "true"
+          pullRequests: "false"
+          compareLink: "true"
+          issuesWoLabels: "false"
+          futureRelease: "${{ inputs.new-tag }}"
+          issuesLabel: "**Miscellaneous:**"
+          bugLabels: "type/bug"
+          enhancementLabels: "type/enhancement"
+          httpCache: "true"
+          verbose: "true"
+          addSections: |
+            ${{ needs.read-config-file.outputs.content }}
+
+      - name: Update github Changelog
+        run: |
+          # Remove header from changes
+          sed -i '1d' changes.md
+          # Prepend changes.md after first header
+          sed -i '/# Changelog/ r changes.md' "${{ inputs.gh-changelog }}"
+        shell: bash
+
+      - name: Update doc Changelog
+        run: |
+          if [ -n "${{ inputs.doc-changelog }}" ]; then 
+            sed -i '1d' changes.md
+            sed -i '/# Changelog/ r changes.md' "${{ inputs.doc-changelog }}"
+          fi
+        shell: bash
+
+      - name: Create release commit
+        run: |
+          # Set email and username for github-actions bot
+          git config --local user.email "${{ secrets.github-email }}"
+          git config --local user.name "${{ secrets.github-token }}"
```

This leaks the token!

```suggestion
          git config --local user.name "${{ secrets.github-username }}"
```

---

Thanks @torbsto, I replaced the token with a new one. @MichaelKora please be careful where you set that token